script.py

```diff
@@ -1,37 +1,62 @@
 import datetime
 import hashlib
+import requests
 
 
-def totimestamp(dt, epoch=datetime.datetime(1970,1,1)):
+def totimestamp(dt, epoch=datetime.datetime(1970, 1, 1)):
     td = dt - epoch
     # return td.total_seconds()
-    return (td.microseconds + (td.seconds + td.days * 86400) * 10**6) / 10**6 
+    return (td.microseconds + (td.seconds + td.days * 86400) * 10**6) / 10**6
 
-streamername = input("Enter streamer name: ").strip()
-vodID = input("Enter vod id: ").strip()
-timestamp = input("Enter VOD timestamp (YYYY-MM-DD HH:MM:SS) UTC:  ").strip()
 
-year = int(timestamp[:4])
-month = int(timestamp[5:7])
-day = int(timestamp[8:10])
+def geturl(year, month, day, hour, minute, seconds):
 
-hour = int(timestamp[11:13])
-minute = int(timestamp[14:16])
-seconds = int(timestamp[17:])
+    td = datetime.datetime(year, month, day, hour, minute, seconds)
+
+    converted_timestamp = totimestamp(td)
 
-td = datetime.datetime(year,month,day,hour,minute,seconds)
+    formattedstring = streamername + "_" + vodID + "_" + str(int(converted_timestamp))
 
+    hash = str(hashlib.sha1(formattedstring.encode('utf-8')).hexdigest())
 
-converted_timestamp = totimestamp(td)
+    requiredhash = hash[:20]
 
-formattedstring = streamername + "_" + vodID + "_" + str(int(converted_timestamp))
+    finalformattedstring = requiredhash + '_' + formattedstring
 
-hash = str(hashlib.sha1(formattedstring.encode('utf-8')).hexdigest())
+    url = f"https://vod-secure.twitch.tv/{finalformattedstring}/chunked/index-dvr.m3u8"
 
-requiredhash = hash[:20]
+    return url
 
-finalformattedstring = requiredhash + '_' +  formattedstring
 
-url = f"https://vod-secure.twitch.tv/{finalformattedstring}/chunked/index-dvr.m3u8"
+streamername = input("Enter streamer name: ").strip()
+vodID = input("Enter vod id: ").strip()
+timestamp = input("Enter VOD timestamp (YYYY-MM-DD HH:MM:SS) UTC:  ").strip()
 
-print(url)
+year = int(timestamp[:4])
+month = int(timestamp[5:7])
+day = int(timestamp[8:10])
+
+hour = int(timestamp[11:13])
+minute = int(timestamp[14:16])
+seconds = timestamp[17:]
+
+if seconds == '--':
+    print('searching ', end='', flush=True)
+    for seconds in range(60):
+        print('.', end='', flush=True)
+        url = geturl(year, month, day, hour, minute, seconds)
+        data = requests.get(url)
+        found = 0
```

Comment:
Feels kinda spammy to, at the worst case, make 60 different get requests, just to get the right timestamp with seconds